core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -753,21 +751,25 @@ class HttpGoogleServicesDAO(
     } )
   }
 
-  override def getGenomicsOperation(jobId: String): Future[Option[JsObject]] = {
-
-    import spray.json._
-
+  override def getGenomicsOperation(jobId: String): Future[JsObject] = {
     val opId = s"operations/$jobId"
-
     val genomicsApi = new Genomics.Builder(httpTransport, jsonFactory, getGenomicsServiceAccountCredential).setApplicationName(appName).build()
     val operationRequest = genomicsApi.operations().get(opId)
 
-    retryWithRecoverWhen500orGoogleError[Option[JsObject]](() => {
-      // Google library returns a Map[String,AnyRef], but we don't care about understanding the response
-      // So, use Google's functionality to get the json string, then parse it back into a generic json object
-      Some(executeGoogleRequest(operationRequest).toPrettyString.parseJson.asJsObject)
-    }) {
-      case t: HttpResponseException if t.getStatusCode == StatusCodes.NotFound.intValue => None
+    // This predicate is true when it's a 500 or Google error but NOT a Google 404.
+    //
+    // This uses the "applicative builder" syntax from cats to combine 2 predicates
+    // (when500orGoogleError and whenGoogle404) using a binary operator (e.g.
+    // (Boolean, Boolean) => Boolean), and return a resulting predicate.
+    val predicate = (when500orGoogleError |@| whenGoogle404).map((a, b) => a && !b)
```

Comment:
I don't have a problem with the `|@|`, but I'd rename `predicate` to something a bit more descriptive.

I still think this is less clear to users. You're treating 404 as a failure when it's an expected return status for the use case. The user will get whatever Google responds with, and it'll be caught by `PerRequest.logResponse` and logged as a warning. I don't think either of these are desirable; there's a lot to be said for transforming expected errors into something user-readable.